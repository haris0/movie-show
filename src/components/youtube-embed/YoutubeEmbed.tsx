
interface props {
  embedid?: string;
};

const YoutubeEmbed = ({ embedid }: props) => {
  if (!embedid) return null;

  return (
    <div className="relative h-0 overflow-hidden pb-[56.25%]">
      <iframe
        className="absolute inset-0 w-full h-full"
        src={`https://www.youtube.com/embed/${embedid}`}
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
        title="Embedded youtube"
      />
    </div>
  )
};

export default YoutubeEmbed;
