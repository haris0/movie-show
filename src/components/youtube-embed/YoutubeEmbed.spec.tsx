import { render } from '@testing-library/react';
import YoutubeEmbed from './YoutubeEmbed';

describe('YoutubeEmbed Component', () => {
  it('should render correctly', () => {
    const { baseElement } = render(
      <YoutubeEmbed embedid=''/>,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
