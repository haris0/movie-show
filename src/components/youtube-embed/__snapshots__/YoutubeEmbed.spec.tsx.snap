// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`YoutubeEmbed Component should render correctly 1`] = `
<body>
  <div />
</body>
`;
